pkg/issuer/acme/issue.go

```diff
@@ -46,9 +46,9 @@ func (a *Acme) obtainCertificate(ctx context.Context, crt *v1alpha1.Certificate)
 	}
 
 	// get existing certificate private key
-	key, err := kube.SecretTLSKey(a.secretsLister, crt.Namespace, crt.Spec.SecretName)
+	key, err := kube.SecretRSAKey(a.secretsLister, crt.Namespace, crt.Spec.SecretName)
 	if k8sErrors.IsNotFound(err) || errors.IsInvalidData(err) {
-		key, err = pki.GenerateRSAPrivateKey(2048)
+		key, err = pki.GenerateRSAPrivateKey(pki.MinRSAKeySize)
```

Comment:
Should this be GeneratePrivateKeyForCertificate?

Reply:
& a change to use `PublicKeyForPrivateKey` and `EncodePrivateKey`

Reply:
I actually left that there specifically as acme is rsa only at the moment. You're right though, it does make sense to use `GeneratePrivateKeyForCertificate` in that block so when acme protocol supports ecdsa, there'll be no change needed.